src/global_context.rs

```diff
@@ -60,6 +66,7 @@ pub struct GlobalContext {
     pub telemetry: Arc<StdRwLock<telemetry_structs::Storage>>,
     pub vecdb_search: Arc<AMutex<Box<dyn VecdbSearch + Send>>>,
     pub ask_shutdown_sender: Arc<Mutex<std::sync::mpsc::Sender<String>>>,
+    pub lsp_backend_data: LSPBackendData,
```

Comment:
Name must be more representative. I see lsp_backend_data and don't understand for what it